google-cloud-storage/lib/google/cloud/storage/policy/bindings.rb

```diff
@@ -0,0 +1,184 @@
+        #   bucket.uniform_bucket_level_access = true
+        #
+        #   bucket.policy requested_policy_version: 3 do |p|
+        #     p.version # the value is 1
+        #     p.version = 3 # Must be explicitly set to opt-in to support for conditions.
+        #
+        #     expr = "resource.name.startsWith(\"projects/_/buckets/bucket-name/objects/prefix-a-\")"
+        #     p.bindings.insert({
+        #                         role: "roles/storage.admin",
+        #                         members: ["user:owner@example.com"],
+        #                         condition: {
+        #                           title: "my-condition",
+        #                           description: "description of condition",
+        #                           expression: expr
+        #                         }
+        #                       })
+        #   end
+        #
+        class Bindings
+          include Enumerable
+
+          ##
+          # @private Creates a Bindings object.
+          def initialize
+            @bindings = []
+          end
+
+          ##
+          # Adds a binding or bindings to the collection. The arguments may be
+          # {Google::Cloud::Storage::Policy::Binding} objects or equivalent hash
+          # objects that will be implicitly coerced to binding objects.
+          #
+          # @param [Google::Cloud::Storage::Policy::Binding, Hash] bindings One
+          #   or more bindings to be added to the policy owning the collection.
+          #   The arguments may be {Google::Cloud::Storage::Policy::Binding}
+          #   objects or equivalent hash objects that will be implicitly coerced
+          #   to binding objects.
+          #
+          # @example Updating a Policy from version 1 to version 3:
+          #   require "google/cloud/storage"
+          #
+          #   storage = Google::Cloud::Storage.new
+          #   bucket = storage.bucket "my-bucket"
+          #
+          #   bucket.uniform_bucket_level_access = true
+          #
+          #   bucket.policy requested_policy_version: 3 do |p|
+          #     p.version # the value is 1
+          #     p.version = 3 # Must be explicitly set to opt-in to support for conditions.
+          #
+          #     expr = "resource.name.startsWith(\"projects/_/buckets/bucket-name/objects/prefix-a-\")"
+          #     p.bindings.insert({
+          #                         role: "roles/storage.admin",
+          #                         members: ["user:owner@example.com"],
+          #                         condition: {
+          #                           title: "my-condition",
+          #                           description: "description of condition",
+          #                           expression: expr
+          #                         }
+          #                       })
+          #   end
+          #
+          def insert *bindings
+            bindings = coerce_bindings(*bindings)
+            @bindings += bindings
```

Comment:
Could we have `insert` and `remove` both return `self` instead of the internal `@bindings` array?

Reply:
I was wondering about that. What do you think about `remove` somehow indicating whether the operation was successful? Maybe both methods should return the added or removed bindings?

Reply:
@blowmage What do you think ^

Reply:
I've seen it done both ways. (Even the Ruby standard classes are inconsistent. e.g. `Array#delete(obj)` and `Array#delete_at(index)` both return the object deleted, but `Array#delete_if { ... }` returns `self`.) Personally I prefer to return `self` consistently from this kind of mutation method because I like chaining. But I'm open to other opinions.

Reply:
I'll update to `self` now to fix the leak. @blowmage: Feel free to request a different solution!

Reply:
@dazuma Updated to return `self`.